### PR TITLE
Added getters for EmbedBuilder

### PR DESCRIPTION
**Pull request by VigusWG**
## Pull Request Etiquette

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [X] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Added to EmbedBuilder
- getTitle()
- getFooter()
- getTimestamp()
- getColor()
- getThumbnail()
- getImage()
- getAuthor()

Currently there is no other way to get them.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 1]
```diff
@@ -386,6 +409,17 @@ public EmbedBuilder setTimestamp(@Nullable TemporalAccessor temporal)
         return this;
     }
 
+    /**
+     * The {@link java.awt.Color Color} of the embed
+     *
+     * @return Color the builder has been set with.
+     */
+    @Nonnull
+    public java.awt.Color getColor()
```
Comment (duncte123) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:418]: You should also add `getColorRaw`, keep it in line with the rest of JDA

**Comment by VigusWG** [comment]
ps sorry if it didnt do everything right, this is my first PR ever :)

**Review by duncte123 — CHANGES_REQUESTED** [review]

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java (hunk 1 above)
Comment (VigusWG) [reply src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:418]: thanks

Comment (Chew) [reply src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:418]: We already import java.awt.Color so we don't need to use FQDN

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 2]
```diff
@@ -521,6 +590,18 @@ public EmbedBuilder setImage(@Nullable String url)
         return this;
     }
 
+    /**
+     * The {@link net.dv8tion.jda.api.entities.MessageEmbed.AuthorInfo AuthorInfo} used to
+     * build the footer for the embed.
```
Comment (Chew) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:595]:
```suggestion
     * build the author for the embed.
```

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 3]
```diff
@@ -386,6 +409,28 @@ public EmbedBuilder setTimestamp(@Nullable TemporalAccessor temporal)
...
+    /**
+     * The {@link Int int} of the color as a raw integer
+     *
+     * @return Raw color the builder has been set with.
+     */
+    @Nonnull
+    public int getColorRaw()
+    {
+        return color;
+    }
```
Comment (Xirado) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:432]: You shouldn't annotate primitives with nullability annotations

Comment (VigusWG) [reply src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:432]: Alright will fix, thanks.

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java (hunk 1 above)
Comment (VigusWG) [reply src/main/java/net/dv8tion/jda/api/EmbedBuilder.java]: All fixed, thanks

**Comment by Xirado** [comment]
You should also check what those getters actually return. I don't think they should be `@Nonnull`

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 4]
```diff
@@ -425,6 +469,18 @@ public EmbedBuilder setColor(int color)
         return this;
     }
 
+    /**
+     * The {@link net.dv8tion.jda.api.entities.MessageEmbed.Thumbnail Thumbnail} used to
+     * build the thumbnail for the embed.
+     *
+     * @return Thumbnail with current thumbnail context.
+     */
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:477]:
```suggestion
    /**
     * The information about the {@link net.dv8tion.jda.api.entities.MessageEmbed.Thumbnail Thumbnail} image to be displayed with the embed.
     * <br>If a {@link net.dv8tion.jda.api.entities.MessageEmbed.Thumbnail Thumbnail} hasn't been set on this builder, this returns null.
     *
     * @return Possibly-null {@link net.dv8tion.jda.api.entities.MessageEmbed.Thumbnail Thumbnail} instance
     *         containing general information on the displayable thumbnail.
     */
```

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 5]
```diff
@@ -472,6 +528,18 @@ public EmbedBuilder setThumbnail(@Nullable String url)
         return this;
     }
 
+    /**
+     * The {@link net.dv8tion.jda.api.entities.MessageEmbed.ImageInfo ImageInfo} used to
+     * build the image for the embed.
+     *
+     * @return ImageInfo with current image context.
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:535]:
```suggestion
     * The information about the image in builder, or null when not set
     *
     * @return Possibly-null {@link net.dv8tion.jda.api.entities.MessageEmbed.ImageInfo ImageInfo}
     *         containing image information.
```

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 6]
```diff
@@ -521,6 +589,18 @@ public EmbedBuilder setImage(@Nullable String url)
         return this;
     }
 
+    /**
+     * The {@link net.dv8tion.jda.api.entities.MessageEmbed.AuthorInfo AuthorInfo} used to
+     * build the author for the embed.
+     *
+     * @return AuthorInfo with current author context
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:596]:
```suggestion
     * The information on the author of builder.
     * <br>This is typically used to represent the account on the providing site.
     *
     * @return Possibly-null {@link net.dv8tion.jda.api.entities.MessageEmbed.AuthorInfo AuthorInfo}
     *         containing author information.
```

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 7]
```diff
@@ -633,6 +713,18 @@ public EmbedBuilder setAuthor(@Nullable String name, @Nullable String url, @Null
         return this;
     }
 
+    /**
+     * The {@link net.dv8tion.jda.api.entities.MessageEmbed.Footer Footer} used to
+     * build the footer for the embed.
+     *
+     * @return Footer with current footer context.
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:720]:
```suggestion
     * The footer (bottom) of the embedded content.
     * <br>This is typically used for timestamps or site icons.
     *
     * @return Possibly-null {@link net.dv8tion.jda.api.entities.MessageEmbed.Footer Footer}
     *         containing the embed footer content.
```

**Review by Tais993 — COMMENTED** [review]
Missing getter for fields, description and the title URL, annotations should be changed to `@Nullable` and I'd personally keep the javadoc almost equal to the MessageEmbed documentation.

No need for differences, can be the same

**Comment by VigusWG** [comment]
Thanks everyone for your critiques, I apologies for not having everything perfect the first time.

**Comment by Andre601** [comment]
There are still some inconsistencies imo.

For example are there lines like `@return Nullable <something>` while other lines have `@return Possibly-null <something>`

I think you should stick with the second one as that is afaik more used in the docs and the first one makes more sense for `@param` descriptions of nullable ones.

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 8]
```diff
@@ -366,6 +377,17 @@ public EmbedBuilder appendDescription(@Nonnull CharSequence description)
         return this;
     }
 
+    /**
+     * The Timestamp of the builder
+     *
+     * @return Nullable {@link java.time.OffsetDateTime OffsetDateTime} object representing the timestamp
+     */
+    @Nonnull
```
Comment (freya022) [inline src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:385]: Should be Nullable

**Review by freya022 — CHANGES_REQUESTED** [review]

**Comment by VigusWG** [comment]
> 

These should now all been fixed.

**Review by freya022 — CHANGES_REQUESTED** [review]

File: src/main/java/net/dv8tion/jda/api/EmbedBuilder.java [hunk 9]
```diff
@@ -472,6 +531,18 @@ public EmbedBuilder setThumbnail(@Nullable String url)
         return this;
     }
 
+    /**
+     * The information about the image in builder, or null when not set
+     *
+     * @return Possibly-null {@link net.dv8tion.jda.api.entities.MessageEmbed.ImageInfo ImageInfo}
+     *         containing image information.
+     */
+    @Nonnull
```
[the remark this answers is not in the document]
Comment (VigusWG) [reply src/main/java/net/dv8tion/jda/api/EmbedBuilder.java:540]: idk how i missed that lol.
all fixed

**Comment by MinnDevelopment** [comment]
> Currently there is no other way to get them.

You can use `build().getX()`, I don't think this is needed. Builders in JDA usually do not offer getters because the setters are asymmetric to the getters. For instance, when you do `setTitle(name, url)`, you cannot make a getter that returns both unless its wrapped into a specific type, which is not symmetric and kind of defeats the idea of setter/getter relationships.

**Comment by VigusWG** [comment]
> > Currently there is no other way to get them.
> 
> You can use `build().getX()`, I don't think this is needed. Builders in JDA usually do not offer getters because the setters are asymmetric to the getters. For instance, when you do `setTitle(name, url)`, you cannot make a getter that returns both unless its wrapped into a specific type, which is not symmetric and kind of defeats the idea of setter/getter relationships.

Fair enough just sounds a little inefficient, but I understand the main point of it. Apologies for my java inexperience, I'll close.

**Pull request closed without merging**